python/paddle/fluid/layers/control_flow.py

```diff
@@ -1797,6 +1797,131 @@ def cond(pred, true_fn=None, false_fn=None, name=None):
     return merged_output
 
 
+def _error_message(what, arg_name, op_name, right_value, error_value):
```

Comment:
I saw you raise TypeError in _error_message and also raise at where you use _error_message, that's not good.

Two options:

Don't raise TypeError in _error_message, but return the error message as a string in _error_message

Remove "raise TypeError" here, rename "_error_message" to raise_error_with_message or some other name properly.

Reply:
Thanks. Done according to option 1.